File: 227-basic-calculator-ii/basic-calculator-ii.py
class Solution:
    def calculate(self, str: str) -> int:

        n = len(str)
        stack = []
        opr = '+'
        curNum = 0

        for i in range(n):
            ch = str[i]
            if ch.isdigit():
                curNum = curNum*10 + int(ch)
            
            if (not ch.isdigit() and ch != ' ') or i == n-1:
                print(ch)
                if opr == '+':
                    # print('+')
                    stack.append(curNum)
                elif opr == '-':
                    # print('e-')
                    stack.append(-curNum)
                elif opr == '*':
                    # print('*')
                    stack.append(stack.pop() * curNum)
                elif opr == '/':
                    # print('/')
                    stack.append(int(stack.pop() / curNum))

                curNum = 0
                opr = ch
        
        result = 0
     
        while stack:
            result += stack.pop()

        return result

            

        

File: 227-basic-calculator-ii/basic-calculator-ii.cpp
class Solution {
public:
    int calculate(string str) {
        int n= str.size();
        int lastNum = 0;
        int curNum = 0;
        char opr = '+';
        int result = 0;
        for(int i=0;i<n;i++){
            if(str[i] >= '0' and str[i] <= '9'){
                curNum = curNum*10 + (str[i] - '0');
            }
            if(!isdigit(str[i]) && str[i] != ' ' or i == (n-1)){
                char curOpr = str[i];
                if(opr == '+'){
                    result += lastNum;
                    lastNum = curNum;
                }
                else if(opr == '-'){
                    result += lastNum;
                    lastNum = -curNum;
                }
                else if(opr == '*'){
                    lastNum =  lastNum * curNum;
                    
                }
                else if(opr == '/'){
                    lastNum = lastNum / curNum;
                }
                curNum = 0;
                opr = curOpr;
            }

        }
        result += lastNum;
        return result;
    }
};